### PR TITLE
RHIROS-492 upgrade react-router-dom to v6

### PR DESCRIPTION
**Pull request by r14chandra**
## Why do we need this change? :thought_balloon:

[RHIROS-492](https://issues.redhat.com/browse/RHIROS-492) 

## Documentation requires update? :memo:

- [ ] Yes
- [X] No

## :guardsman: Checklist :dart:

- [ ] Bugfix
- [ ] New Feature
- [X] Refactor
- [ ] Unittests Added
- [ ] DRY code
- [X] Dependency Added

## Additional :mega:

### CODE REVIEW

**Comment by PreetiW** [comment]
@r14chandra Can you please rebase this PR so it stays upto date with latest master changes, thanks! 🙌🏻

**Comment by gkarat** [comment]
Please keep in mind that you also need to bump @redhat-cloud-services/frontend-components to the version at least 3.10.0 and @redhat-cloud-services/frontend-components-utilities to at least 3.6.0.

These versions provide a compatible layer for InventoryTable that will help you avoid regressions while releasing to production disregarding if Inventory is released or not: https://github.com/RedHatInsights/frontend-components/pull/1744.

**Review by gkarat — CHANGES_REQUESTED** [review]
Please keep in mind that you also need to bump @redhat-cloud-services/frontend-components to the version at least 3.10.0 and @redhat-cloud-services/frontend-components-utilities to at least 3.6.0.

These versions provide a compatible layer for InventoryTable that will help you avoid regressions while releasing to production disregarding if Inventory is released or not: https://github.com/RedHatInsights/frontend-components/pull/1744.

File: package.json
Comment (gkarat) [inline package.json:1]: I hope so many dependencies upgrades within this PR won't cause any bugs. I would limit the upgrades only to those that are required at the moment (for this PR).

**Review by gkarat — APPROVED** [review]
Looks good to me! Tested locally and couldn't spot any issues. Thanks @r14chandra

File: package.json
Comment (r14chandra) [reply package.json:1]: Thanks for the review @gkarat. It was added by Martin. I will revert the dependencies upgrades.

**Comment by PreetiW** [comment]
@r14chandra can you please check why the tests are failing?

**Comment by r14chandra** [comment]
> @r14chandra can you please check why the tests are failing?

Tests are not failing, it is failing on the deployment job.  
I have asked this on the channel.

**Comment by Hyperkid123** [comment]
@r14chandra you are missing a babel dependency. Run `npm i --save-dev @babel/runtime` and the issue will go away.

**Comment by PreetiW** [comment]
@r14chandra can you please rebase this branch?

**Comment by yash2189** [comment]
Tested this on EE and executed iqe smoke tests for ros plugin. detailed observations are added on Jira

**Pull request merged**